### PR TITLE
feat: implement inverse fast Fourier transform (irfftn) function for paddle frontend

### LINKED ISSUE #22913: irfftn

### PR DESCRIPTION
**Pull request by xingshuodong**
# PR Description 
Feat: implement inverse fast Fourier transform (irfftn) function for paddle frontend

Add paddle.fft Fast Fourier Transform Functions to Paddle Frontend #15047 
This pull request is related to resolves [irfftn #22913].
- [x] <irfftn> #22913 
Close #22913 
## Checklist 

- [x] Did you add a function?
- [x] Did you add the tests?
- [x] Did you follow the steps we provided?

### Socials: 
xingshu@deakin.edu.au
ins: xingshuodong

### CODE REVIEW

**Comment by xingshuodong** [comment]
need fix function

**Comment by xingshuodong** [comment]
Some questions, if there is no irfft or similar function in backends now, am i suppose to use different way to write frontend function achieve the irfftn, i am very new and please guide me to do so.

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by xingshuodong** [comment]
@tomatillos Hi Tom, can you help to check why here hfft frontend function and irfft frontend function can not pass the test locally regardless the irfftn function.

**Comment by xingshuodong** [comment]
@ivy-seed can assign me a new reviewer? Tom is not avaliable to contact

**Comment by tomatillos** [comment]
Hi, your task is only to add the `irfftn` function, so don't worry about the other failing ones. It would be best if it supported more than n dimensions though

**Comment by xingshuodong** [comment]
> Hi, your task is only to add the `irfftn` function, so don't worry about the other failing ones. It would be best if it supported more than n dimensions though

Hi Tom, appreciate let me know that, so currently it's fine right? @tomatillos

**Comment by tomatillos** [comment]
Nearly, can you make sure the test passes with the multidimensional version as well? At the moment you're only testing for n=1

**Comment by xingshuodong** [comment]
Sure Tom

**Comment by xingshuodong** [comment]
@tomatillos Hi Tom, 
I may need help to properly develop N-dim function for irfftn, it is better to have backend core function of irfftn to directly manuplate the along the N-dim.

If you are happy you can take my updated irfftn which is working for 1 D for numpy tensorflow and paddlem as part of solution which will enable me to apply intern.

If you require a complete develop, and yes it is possible to achieve the true irfftn, but I may need more time to solve it which a bit out of plan.

**Comment by xingshuodong** [comment]
> Hi, we'd need full functionality of `irfftn` to get this merged (i.e. for multidimensional case too). You are welcome to choose another task to work on :)

Hi Tom i will achieve full functionality in this week, probably need go through to see what happen in function, if it is super complex and need a range of functions support, then yea I need pick another task

**Comment by xingshuodong** [comment]
@tomatillos  Hi, Tom, please review the code, the current one is working for n-dim situation by setting min dim and max dim, 
also it works work any dim to pass the test, it is very or most close to a true irfftn.

To achieve a flexibility use of irfftn
-[ ] specific n-th dim
-[ ] a range of dim like from 2 to 5  is this necessary needed?

the potential issue is, the backend of ifft maybe has some issue, while ifftn is working well as its logic go to last axis, 
then we have to modify each axis as last axis to get rid of impacting other axis in the case, if you can't understand what i am saying it may be my issue report is not clear, but for now, the irfftn is working as expected.

**Review by Kilvny — COMMENTED** [review]
how did you choose another task

**Comment by xingshuodong** [comment]
> how did you choose another task



> how did you choose another task

what do you mean by choosing another task

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_fft.py [hunk 1]
```diff
@@ -242,3 +242,39 @@ def test_paddle_rfftfreq(
...
+    input_dtypes, x, axis = dtype_x_axis
+    helpers.test_frontend_function(
+        input_dtypes=input_dtypes,
+        backend_to_test=backend_fw,
+        frontend=frontend,
+        test_flags=test_flags,
+        fn_tree=fn_tree,
+        x=x[0],
+        s=None,
+        axes=[-1],
```
Comment (tomatillos) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_fft.py:278]: Why are you only testing axis=-1, you should pass axes=axis here.

**Pull request closed without merging**